src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -102,6 +102,36 @@ UA_Server_addWriterGroup(UA_Server *server, const UA_NodeId connection,
             &UA_TYPES[UA_TYPES_UADPWRITERGROUPMESSAGEDATATYPE];
         newConfig->messageSettings.encoding = UA_EXTENSIONOBJECT_DECODED;
     }
+    /* transportSettings */
+    newWriterGroup->channel = NULL;
+    const UA_ExtensionObject *transportSettings = &writerGroupConfig->transportSettings;
+    if(transportSettings && transportSettings->content.decoded.type == &UA_TYPES[UA_TYPES_DATAGRAMWRITERGROUPTRANSPORT2DATATYPE]) {
+        UA_DatagramWriterGroupTransport2DataType *ts = (UA_DatagramWriterGroupTransport2DataType *) transportSettings->content.decoded.data;
+
+        if(ts->address.content.decoded.type == &UA_TYPES[UA_TYPES_NETWORKADDRESSURLDATATYPE]) {
+            UA_NetworkAddressUrlDataType *address = (UA_NetworkAddressUrlDataType *) ts->address.content.decoded.data;
+            const char *prefix = "opc.udp://";
+            if(strncmp(prefix, (char *) address->url.data, strlen(prefix)) == 0) {
+
+                UA_LOG_DEBUG(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                             "adding udp unicast WriterGroup connecting to "
+
+                             );
+                /* Retrieve the transport layer for the given profile uri */
+                UA_PubSubTransportLayer *tl = UA_getTransportProtocolLayer(server, &currentConnectionContext->config->transportProfileUri);
+                UA_CHECK_MEM_ERROR(tl, return UA_STATUSCODE_BADNOTFOUND, &server->config.logger,
+                                   UA_LOGCATEGORY_SERVER, "PubSub Connection creation failed. Requested transport layer not found.");
+                UA_TransportLayerContext ctx;
+                ctx.writerGroupAddress = address;
+                ctx.connection = currentConnectionContext;
+                ctx.connectionConfig = currentConnectionContext->config;
+                ctx.decodeAndProcessNetworkMessage = UA_decodeAndProcessNetworkMessage;
+                ctx.server = server;
+
+                newWriterGroup->channel = tl->createPubSubChannel(tl, &ctx);
+            }
```

Comment:
There should also be a check if the Connection->address is opc.udp://localhost:[port] in case the configuration is invalid